src/components/complex/Proposal/index.jsx

```diff
@@ -239,6 +243,87 @@ const Proposal = ({
     }
   }, [noData, activeChainId])
 
+  const handleMessageFromIframe = (event) => {
+    // Check the origin of the message for security
+    if (event.origin !== window.location.origin) {
+      console.error('Received an untrusted message from:', event.origin)
+      return
+    }
+
+    // Log or handle the received message
+    if (event.data.type === 'linkClick') {
+      if (event.data.id !== id) return
+      setReadMoreContent(event.data.href)
+      setLocation({ current: event.data.href, old: location.current })
+    }
+  }
+
+  const goBack = () => {
+    setReadMoreContent(location.old)
+    setLocation({ current: location.old, old: url })
+  }
+
+  useEffect(() => {
+    window.addEventListener('message', handleMessageFromIframe)
+
+    return () => {
+      window.removeEventListener('message', handleMessageFromIframe)
+    }
+  }, [])
+
+  useEffect(() => {
+    // // Function to perform search and replace within the iframe
+    const performSearchAndReplace = async () => {
+      if (readMoreContent && readMoreContent != 'isLoading') {
+        const html = await (await fetch(readMoreContent)).text()
+        const restoredPoundHTML = html.replace(/%23/g, '#')
+        const googleLinksRegex = /https:\/\/www\.google\.com\/url\?q=([^&]+)(&sa=|&amp;)[^"]*/g
+        const parser = new DOMParser()
+        const doc = parser.parseFromString(restoredPoundHTML, 'text/html')
+        const ipfsLinks = Array.from(doc.querySelectorAll('a')).filter(
+          (link) => link.getAttribute('href') && link.getAttribute('href').includes('/ipfs/')
+        )
+        ipfsLinks.forEach((link) => {
```

Comment:
```suggestion
        ipfsLinks.map((link) => {
```
Prefer .map if you don't need iteration